src/server/pps/server/pipeline_controller.go

```diff
@@ -203,7 +198,11 @@ func (op *pipelineOp) run() error {
 		}
 		// deletePipelineResources calls cancelMonitor() and cancelCrashingMonitor()
 		// in addition to deleting the RC, so those calls aren't necessary here.
-		return op.deletePipelineResources()
+		if err := op.deletePipelineResources(); err != nil {
```

Comment:
I actually think we may want to retry deleting the pipeline resources here, if possible (kubernetes might be transiently unavailable). I wrote a suggestion above to change `failPipelineError` to `errRetryStep` that might allow this codepath to be retried in the same way as the others.

If it turns out `errRetryStep` won't work and we can't feasibly retry here, it's not a huge deal because `pollPipelines` should eventually come around and retry itself, but I think `errRetryStep{}` might be an improvement over that, since `pollPipelines` might not come around for a several minutes.

Reply:
I was hoping you could provide some insight here, thanks. It seemed like the previous code went out of its way to not retry this case, and I wasn't sure I wanted to shove more complexity into the error type. Allowing retry makes sense to me